src/reconcile/rec_visibility.c

```diff
@@ -646,8 +677,11 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
          * is the same as it was the previous reconciliation. Otherwise lookaside can end up with
          * two birthmark records in the same update chain.
          */
-        WT_ERR(__rec_append_orig_value(session, page, first_inmem_upd, vpack));
-        upd_select->upd = NULL;
+        WT_ERR(__rec_append_orig_value(session, page, ins, ripcip, first_inmem_upd, vpack));
+        if (upd->ext != 0)
```

Comment:
@sulabhM Could you please double check what I've done here? I find this section of code a bit confusing since we're using both `upd` and `upd_select->upd` and it's not 100% clear to me what we're expecting `upd` to be pointing at a given point in time.